SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc

```diff
@@ -520,35 +601,57 @@ void HGCDigitizer::accumulate(edm::Handle<edm::PCaloHitContainer> const &hits,
 void HGCDigitizer::accumulate(const PHGCSimAccumulator& simAccumulator) {
   //configuration to apply for the computation of time-of-flight
   bool weightToAbyEnergy(false);
-  switch( mySubDet_ ) {
-  case ForwardSubdetector::HGCEE:
-    weightToAbyEnergy = theHGCEEDigitizer_->toaModeByEnergy();
-    break;
-  case ForwardSubdetector::HGCHEF:
-    weightToAbyEnergy = theHGCHEfrontDigitizer_->toaModeByEnergy();
-    break;
-  case ForwardSubdetector::HGCHEB:
-    weightToAbyEnergy = theHGCHEbackDigitizer_->toaModeByEnergy();
-    break;
-  default:
-    break;
+  if (geometryType_ == 0) {
```

Comment:
this pattern is a partial repeat of one from the other `accumulate()` function. It would be nice to reduce the duplication if it can be rearranged efficiently.

Reply:
The `getWeight()` function now introduced should be used here (and in the other `accumulate()` function)